tests/php/Unit/Compiler/Analyzer/AnalyzePersistentListTest.php

```diff
@@ -0,0 +1,249 @@
+    {
+        $list = TypeFactory::getInstance()->persistentListFromArray([
+            Symbol::create(Symbol::NAME_QUOTE),
+             'any text',
+        ]);
+        self::assertInstanceOf(QuoteNode::class, $this->listAnalyzer->analyze($list, NodeEnvironment::empty()));
+    }
+
+    public function testSymbolWithNameDo(): void
+    {
+        $list = TypeFactory::getInstance()->persistentListFromArray([
+            Symbol::create(Symbol::NAME_DO), 1,
+        ]);
+        self::assertInstanceOf(DoNode::class, $this->listAnalyzer->analyze($list, NodeEnvironment::empty()));
+    }
+
+    public function testSymbolWithNameIf(): void
+    {
+        $list = TypeFactory::getInstance()->persistentListFromArray([
+            Symbol::create(Symbol::NAME_IF),
+            true,
+            true,
+        ]);
+        self::assertInstanceOf(IfNode::class, $this->listAnalyzer->analyze($list, NodeEnvironment::empty()));
+    }
+
+    public function testSymbolWithNameApply(): void
+    {
+        $list = TypeFactory::getInstance()->persistentListFromArray([
+            Symbol::create(Symbol::NAME_APPLY), '+', TypeFactory::getInstance()->persistentVectorFromArray(['']),
+        ]);
+        self::assertInstanceOf(ApplyNode::class, $this->listAnalyzer->analyze($list, NodeEnvironment::empty()));
+    }
+
+    public function testSymbolWithNameLet(): void
+    {
+        $list = TypeFactory::getInstance()->persistentListFromArray([
+            Symbol::create(Symbol::NAME_LET),
+            TypeFactory::getInstance()->persistentVectorFromArray([]),
+            TypeFactory::getInstance()->persistentVectorFromArray([]),
+        ]);
+        self::assertInstanceOf(LetNode::class, $this->listAnalyzer->analyze($list, NodeEnvironment::empty()));
+    }
+
+    public function testSymbolWithNamePhpNew(): void
+    {
+        $list = TypeFactory::getInstance()->persistentListFromArray([
+            Symbol::create(Symbol::NAME_PHP_NEW), '',
+        ]);
+        self::assertInstanceOf(PhpNewNode::class, $this->listAnalyzer->analyze($list, NodeEnvironment::empty()));
+    }
+
+    public function testSymbolWithNamePhpObjectCall(): void
+    {
+        $list = TypeFactory::getInstance()->persistentListFromArray([
+            Symbol::create(Symbol::NAME_PHP_OBJECT_CALL), '', Symbol::create(''),
+        ]);
+        self::assertInstanceOf(
+            PhpObjectCallNode::class,
+            $this->listAnalyzer->analyze($list, NodeEnvironment::empty())
+        );
+    }
+
+    public function testSymbolWithNamePhpObjectStaticCall(): void
+    {
+        $list = TypeFactory::getInstance()->persistentListFromArray([
+            Symbol::create(Symbol::NAME_PHP_OBJECT_STATIC_CALL), '', Symbol::create(''),
+        ]);
+        self::assertInstanceOf(
+            PhpObjectCallNode::class,
+            $this->listAnalyzer->analyze($list, NodeEnvironment::empty())
+        );
+    }
+
+    /*public function testSymbolWithNamePhpArrayGet(): void
```

Comment:
What's the status of these commented tests? Do we need to update them? Or are they obsolete and we should remove them?